paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/AddPaymentMethodRequirement.kt

```diff
@@ -0,0 +1,56 @@
+package com.stripe.android.lpmfoundations.paymentmethod
+
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.PaymentMethod
+
+internal enum class AddPaymentMethodRequirement {
+    /** A special case that indicates the payment method is always unsupported by PaymentSheet. */
+    Unsupported {
+        override fun meetsRequirements(metadata: PaymentMethodMetadata): Boolean = false
+    },
+
+    /** Indicates the payment method is unsupported by PaymentSheet when using SetupIntents or SFU. */
+    UnsupportedForSetup {
+        override fun meetsRequirements(metadata: PaymentMethodMetadata): Boolean {
+            return !metadata.hasIntentToSetup()
+        }
+    },
+
+    /** Indicates that a payment method requires shipping information. */
+    ShippingAddress {
+        override fun meetsRequirements(metadata: PaymentMethodMetadata): Boolean {
+            val shipping = (metadata.stripeIntent as? PaymentIntent)?.shipping
+            return shipping?.name != null &&
+                shipping.address.line1 != null &&
+                shipping.address.country != null &&
+                shipping.address.postalCode != null
```

Comment:
Is this valid for all regions we support? I know some regions may not postal codes.

Reply:
It's the logic we have today, I just ported it :/